src/openms/source/ANALYSIS/ID/AccurateMassSearchEngine.cpp

```diff
@@ -1495,10 +1495,14 @@ namespace OpenMS
     // compute theoretical isotope distribution
     IsotopeDistribution iso_dist(form.getIsotopeDistribution(CoarseIsotopeDistribution((UInt)common_size)));
     std::vector<double> theoretical_iso_dist;
-    for (auto iso_it = iso_dist.begin(); iso_it != iso_dist.end(); ++iso_it)
-    {
-      theoretical_iso_dist.push_back(iso_it->getIntensity());
-    }
+    std::transform(
+      iso_dist.begin(),
+      iso_dist.end(),
+      back_inserter(theoretical_iso_dist),
+      [](const IsotopeDistribution::MassAbundance& p)
+      {
+        return p.getIntensity();
+      });
```

Comment:
@timosachsenberg what is our current policy on these constructs? I feel this is much less readable than
```
for (const auto & iso_it : iso_dist) {theoretical_iso_dist.push_back(iso_it.getIntensity());}
```

Reply:
This change addresses @hroest 's previous comment for constness on the for loop. Unfortunately `const auto ` casts `const` the iterator variable and not the iterator contents. Due to the `IsotopeDistribution` 's overloads both const_iterator and iterator are overloaded by `begin()` [here](https://github.com/pati-ni/OpenMS/blob/feature/coarse_isotope_distributions/src/openms/include/OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/IsotopeDistribution.h#L209). So if this is not good enough we should explicitly define the data type or make a change at `IsotopeDistribution`